src/libcore/slice/mod.rs

```diff
@@ -1541,6 +1541,9 @@ impl<T> [T] {
     /// let src = [1, 2, 3, 4];
     /// let mut dst = [0, 0];
     ///
+    /// // Note: the slices must be the same length, so you can slice the
+    /// // source to be the same size. Here we slice the source, four elements,
```

Comment:
This all make sense to me, though I'm a little confused by this phrase:

> so you can slice the source to be the same size

...because in some scenarios, you'll need to slice the destination to match the length of the source.

I think this would be good enough:

> Note: The slices must be the same length, otherwise this method will panic. Here we slice the source, four elements, to two, the same size as the destination slice.

What do you think?

Reply:
I think changing it to "so you can slice the source or the destination to be the same size." would be fine.